### PR TITLE
Automate release version generation [v8]

### PR DESCRIPTION
**Pull request by gururajsh**
While releasing CLI, take the input from the user for minor/patch bump. Based on the selection, calculate the next release version. New build version need not be stored in a file going forward.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,7 +3,6 @@ NODES ?= 10
 PACKAGES ?= api actor command types util version integration/helpers
 LC_ALL = "en_US.UTF-8"
 
-CF_BUILD_VERSION ?= $$(cat BUILD_VERSION) # TODO: version specific
```
Comment (a-b) [inline Makefile:6]: question(non-blocking): do we still see some value in keeping [build_data.yml](https://github.com/cloudfoundry/cli/blob/main/build_data.yml) up to date? As it was introduced as an eventual alternative to BUILD_VERSION.

**Review by a-b — DISMISSED** [review]

File: bin/bump-version
Comment (a-b) [inline bin/bump-version:1]: praise: yay, removing old code!

File: Makefile (hunk 1 above)
Comment (gururajsh) [reply Makefile:6]: Good point, even that file can be removed. It was mostly used for min capi tests. Since we don't have any, will remove that as well

Comment (gururajsh) [reply Makefile:6]: Done

**Review by joaopapereira — APPROVED** [review]
LGTM

**Pull request merged**